cmake/Pure32.cmake

```diff
@@ -0,0 +1,18 @@
+option (ENABLE_PURE32 "Use only 32-bit types or smaller" OFF)
+option (ENABLE_RASPBERRYPI "Build a library version that's safe for the Raspberry Pi 3" OFF)
+
+if (ENABLE_RASPBERRYPI)
+    set(ENABLE_PURE32 ON)
+    target_compile_definitions(qrack PUBLIC ENABLE_RASPBERRYPI=1)
+endif(ENABLE_RASPBERRYPI)
+
+if (ENABLE_PURE32)
+    set(ENABLE_COMPLEX_X2 OFF)
+    set(ENABLE_COMPLEX8 ON)
+    target_compile_definitions (qrack PUBLIC ENABLE_PURE32=1)
+endif (ENABLE_PURE32)
+
+message ("Raspberry Pi support is: ${ENABLE_PURE32}")
+message ("Pure 32-bit compilation is: ${ENABLE_PURE32}")
+message ("Single accuracy is: ${ENABLE_COMPLEX8}")
+message ("Complex_x2/AVX Support is: ${ENABLE_COMPLEX_X2}")
```

Comment:
All of these messages should not be in the Pure32.cmake file, but instead either at the top level or some other general location.

Reply:
We can put them in the top level, but they depend both on the underlying flags, and whether they're overridden by ENABLE_PURE32 or ENABLE_RASPBERRYPI.